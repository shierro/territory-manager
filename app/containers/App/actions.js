/*
 *
 * MapPage actions
 *
 */

import {
  SET_TOKEN,
  LOGOUT,
  TOGGLE_DRAWER,
} from './constants';

export function setToken(token) {
  return {
    type: SET_TOKEN,
    token,
  };
}

export function logout() {
  return {
    type: LOGOUT,
  };
}

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER,
  };
}
