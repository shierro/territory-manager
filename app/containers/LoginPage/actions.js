/*
 *
 * LoginPage actions
 *
 */

import {
  LOGIN,
  SET_ERROR,
} from './constants';

export function doLogin(username, password) {
  return {
    type: LOGIN,
    username,
    password,
  };
}

export function setError(error) {
  return {
    type: SET_ERROR,
    error,
  };
}
