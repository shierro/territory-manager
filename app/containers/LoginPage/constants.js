/*
 *
 * LoginPage constants
 *
 */

export const LOGIN = 'app/LoginPage/LOGIN';
export const SET_ERROR = 'app/LoginPage/SET_ERROR';
