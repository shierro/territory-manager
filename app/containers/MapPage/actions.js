import {
  GET_INITIAL_LOCATION,
  ADD_PERSON_START,
  SAVE_PERSON_DATA,
  PERSON_FORM_CHANGE,
  PERSON_UPDATE,
  UPDATE_NEW_MARKER_LOCATION,
  SET_STEP,
  CANCEL_ADD,
  PERSON_CLICK,
  SAVE_VISIT,
  TOGGLE_ADDING_VISIT,
  POPUP_CLOSE,
} from './constants';

export function setInitialLocation() {
  return { type: GET_INITIAL_LOCATION };
}

export function addPersonStart() {
  return { type: ADD_PERSON_START };
}

export function savePersonData() {
  return { type: SAVE_PERSON_DATA };
}

export function handleFormChange(key, value) {
  return { type: PERSON_FORM_CHANGE, key, value };
}

export function handlePersonUpdate(key, value) {
  return { type: PERSON_UPDATE, key, value };
}

export function handleNewPersonPositionChange(data) {
  return { type: UPDATE_NEW_MARKER_LOCATION, data };
}

export function moveToStep(step) {
  return { type: SET_STEP, step };
}

export function cancelAdd() {
  return { type: CANCEL_ADD };
}

export function handlePersonClick(index) {
  return { type: PERSON_CLICK, index };
}

export function saveVisit(visitData) {
  return { type: SAVE_VISIT, visitData };
}

export function toggleAddingVisit() {
  return { type: TOGGLE_ADDING_VISIT };
}

export function onPopupClose() {
  return { type: POPUP_CLOSE };
}

export const mapPageActions = dispatch => ({
  setInitialLocation: () => dispatch(setInitialLocation()),
  addPersonStart: () => dispatch(addPersonStart()),
  handleFormChange: (key, value) => dispatch(handleFormChange(key, value)),
  savePersonData: () => dispatch(savePersonData()),
  moveToStep: step => dispatch(moveToStep(step)),
  cancelAdd: () => dispatch(cancelAdd()),
  handleNewPersonPositionChange: data =>
    dispatch(handleNewPersonPositionChange(data)),
  handlePersonUpdate: (key, value) => dispatch(handlePersonUpdate(key, value)),
  handlePersonClick: index => dispatch(handlePersonClick(index)),
  saveVisit: visitData => dispatch(saveVisit(visitData)),
  toggleAddingVisit: () => dispatch(toggleAddingVisit()),
  onPopupClose: () => dispatch(onPopupClose()),
});
