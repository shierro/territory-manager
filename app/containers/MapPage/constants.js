/*
 *
 * MapPage constants
 *
 */

export const DEFAULT_ACTION = 'app/MapPage/DEFAULT_ACTION';
export const SET_INITIAL_LOCATION = 'app/MapPage/SET_INITIAL_LOCATION';
export const GET_INITIAL_LOCATION = 'app/MapPage/GET_INITIAL_LOCATION';
export const SET_PAGE_ERROR = 'app/MapPage/SET_PAGE_ERROR';
