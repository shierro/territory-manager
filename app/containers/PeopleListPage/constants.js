export const SORT = 'app/PeopleListPage/SORT';
export const PAGE_CHANGE = 'app/PeopleListPage/PAGE_CHANGE';
export const ROWS_PER_PAGE_CHANGE = 'app/PeopleListPage/ROWS_PER_PAGE_CHANGE';
