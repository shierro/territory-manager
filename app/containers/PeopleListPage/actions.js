import { SORT, PAGE_CHANGE, ROWS_PER_PAGE_CHANGE } from './constants';

export function handleRequestSort(property) {
  return { type: SORT, property };
}

export function handleChangePage(page) {
  return { type: PAGE_CHANGE, page };
}

export function handleChangeRowsPerPage({ target }) {
  return { type: ROWS_PER_PAGE_CHANGE, rowsPerPage: target.value };
}

export const pageActions = dispatch => ({
  handleRequestSort: (event, property) => dispatch(handleRequestSort(property)),
  handleChangePage: (event, page) => dispatch(handleChangePage(page)),
  handleChangeRowsPerPage: event =>
    dispatch(handleChangeRowsPerPage(event || { target: {} })),
});
