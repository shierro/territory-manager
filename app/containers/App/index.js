/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 * NOTE: while this component should technically be a stateless functional
 * component (SFC), hot reloading does not currently support SFCs. If hot
 * reloading is not a necessity for you then you can refactor it and remove
 * the linting exception.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Switch, Route } from 'react-router-dom';
import { withStyles } from 'material-ui/styles';
import classNames from 'classnames';

import Header from 'components/Header';
import Footer from 'components/Footer';
import LeftDrawer from 'components/LeftDrawer';
import LoginPage from 'containers/LoginPage/Loadable';
import MapPage from 'containers/MapPage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';

import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import {
  makeSelectLocation,
  makeSelectToken,
  makeSelectDrawerOpen,
} from './selectors';
import { logout, toggleDrawer } from './actions';
import reducer from './reducer';
import saga from './saga';
import PrivateRoute from './privateRoute';


const styles = (theme) => ({
  root: {
    flexGrow: 1,
    height: '100vh',
    zIndex: 1,
    overflow: 'hidden',
    position: 'relative',
    display: 'flex',
  },
  toolbar: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    ...theme.mixins.toolbar,
  },
  content: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,
    padding: theme.spacing.unit * 3,
  },
});

class App extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { classes, location, token, drawerOpen } = this.props;
    return (
      <div className={classNames(classes.root, 'root-app')}>
        <Header open={drawerOpen} handleDrawerOpen={this.props.toggleDrawer} token={token} />
        <LeftDrawer
          open={drawerOpen}
          hidden={!token}
          toggleDrawer={this.props.toggleDrawer}
          path={location.path}
          logout={this.props.logout}
        />
        <main className={classes.content}>
          <div className={classes.toolbar} />
          <Switch>
            <Route exact path="/login" component={LoginPage} />
            <Route exact path="/" component={LoginPage} />
            <PrivateRoute exact path="/map" component={MapPage} token={token} />
            <Route component={NotFoundPage} />
          </Switch>
        </main>
        <Footer />
      </div>
    );
  }
}

App.propTypes = {
  classes: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  token: PropTypes.string.isRequired,
  logout: PropTypes.func.isRequired,
  toggleDrawer: PropTypes.func.isRequired,
  drawerOpen: PropTypes.bool.isRequired,
};

const mapStateToProps = createStructuredSelector({
  location: makeSelectLocation(),
  token: makeSelectToken(),
  drawerOpen: makeSelectDrawerOpen(),
});

function mapDispatchToProps(dispatch) {
  return {
    logout: () => dispatch(logout()),
    toggleDrawer: () => dispatch(toggleDrawer()),
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: 'App', reducer });
const withSaga = injectSaga({ key: 'App', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(withStyles(styles, { withTheme: true })(App));
