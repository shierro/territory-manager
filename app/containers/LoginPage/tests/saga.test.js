/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';
import axios from 'axios';
import { push } from 'react-router-redux';

import { LOGIN } from '../constants';
import { setToken } from '../../App/actions';
import { setError } from '../actions';
import defaultSaga, { login, loginFake } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('LoginPage Sagas', () => {
  describe('login Saga', () => {
    let generator;
    beforeEach(() => {
      generator = login({ data: {} });
      const requestURL = 'http://localhost:4000/api/login';
      const callDescriptor = generator.next(axios.get, requestURL).value;
      expect(callDescriptor).toMatchSnapshot();
    });

    it('should dispatch the setToken & redirect if login is successfull', () => {
      const response = 'token';
      const putDescriptor = generator.next(response).value;
      expect(putDescriptor).toEqual(put(setToken(response)));
      const nextRoute = '/map';
      const putDescriptor2 = generator.next(nextRoute).value;
      expect(putDescriptor2).toEqual(put(push(nextRoute)));
    });

    it('should call the setError action if the response errors - generic', () => {
      const response = 'Some error';
      const putDescriptor = generator.throw(response).value;
      expect(putDescriptor).toEqual(
        put(setError('Something went wrong. Try again later')),
      );
    });

    it('should call the setError action if the response errors - Error with message', () => {
      const response = {
        response: { data: { message: 'not enough tickets' } },
      };
      const putDescriptor = generator.throw(response).value;
      expect(putDescriptor).toEqual(put(setError('not enough tickets')));
    });
  });

  describe('loginFake Saga', () => {
    it('should dispatch the setTickets action if it requests the data successfully', () => {
      const generator = loginFake({ username: 'admin', password: 'admin' });
      const callDescriptor = generator.next({}).value;
      expect(callDescriptor).toEqual(put(setToken('testToken')));
      const callDescriptor2 = generator.next('/map').value;
      expect(callDescriptor2).toEqual(put(push('/map')));
    });

    it('should call the setError action if the response errors - generic', () => {
      const response = 'Some error';
      const generator = loginFake({ username: 'wrong', password: 'creds' });
      generator.next();
      const putDescriptor = generator.throw(response).value;
      expect(putDescriptor).toEqual(
        put(setError('Something went wrong. Try again later')),
      );
    });

    it('should call the setError action if the response errors - Error with message', () => {
      const response = {
        response: { data: { message: 'not enough tickets' } },
      };
      const generator = loginFake({ data: {} });
      generator.next();
      const putDescriptor = generator.throw(response).value;
      expect(putDescriptor).toEqual(
        put(setError(response.response.data.message)),
      );
    });
  });

  describe('defaultSaga Saga', () => {
    const watchLoginPageSaga = defaultSaga();
    it('should start task to watch for LOGIN action', () => {
      const takeLatestDescriptor = watchLoginPageSaga.next().value;
      expect(takeLatestDescriptor).toEqual(takeLatest(LOGIN, loginFake));
    });
  });
});
