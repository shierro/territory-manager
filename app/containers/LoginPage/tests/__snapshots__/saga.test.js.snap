// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Sagas login Saga should call the setError action if the response errors - Error with message 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://52791820.ngrok.io/api/login",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`LoginPage Sagas login Saga should call the setError action if the response errors - generic 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://52791820.ngrok.io/api/login",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`LoginPage Sagas login Saga should dispatch the setToken & redirect if login is successfull 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://52791820.ngrok.io/api/login",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
